### PR TITLE
[ShadCN] refactor(icons): use react-icons `IconBase` composition

### PR DESCRIPTION
**Pull request by TylerAPfledderer**

### CODE REVIEW

**Review by pettinarip — COMMENTED** [review]
@TylerAPfledderer this looks great! didn't know we had a `IconBase` in react-icons.

cc @wackerow

**Review by defihook — COMMENTED** [review]
great work
thanks

**Comment by pettinarip** [comment]
Hey @TylerAPfledderer would you need some help on this? its gaining priority now that we need to migrate a bunch of components that use these icons. Would be nice to have this in place to avoid converting all the icons to svgs.

I'll test it locally in the meantime.

**Comment by TylerAPfledderer** [comment]
@pettinarip yes, that would be helpful. Getting bogged down over here at the moment.

I would point out that when double-checking changes to the icons where rendered in components/pages, that if a color was added, it might need to be added to the `fill` css property instead of text color.

`text-*` => `fill-*`

**Comment by pettinarip** [comment]
Ok @TylerAPfledderer.

I've reverted the StakingHierarchy related icons to migrate them later as they are a bit related to that component logic. I don't think we need to address that in this PR.

**Review by pettinarip — COMMENTED** [review]
@TylerAPfledderer I have tested this and it looks good to me. I wonder if you are working on or expecting to do more things in this PR.

Left a comment but I think we should tackle that in a separate PR.

File: src/components/icons/utils.ts [hunk 1]
```diff
@@ -1,7 +1,14 @@
+import type { SVGAttributes } from "react"
 import { IconProps } from "@chakra-ui/react"
 
 export const commonIconDefaultProps: IconProps = {
```
[the remark this answers is not in the document]
Comment (TylerAPfledderer) [reply src/components/icons/utils.ts:4]: @pettinarip Yes, that is correct. Thank you!

**Comment by TylerAPfledderer** [comment]
@pettinarip I have not considered anything additional for this PR.

**Review by pettinarip — APPROVED** [review]
@TylerAPfledderer I fixed a few minor things. Overall I think the icons work well. Thanks for the hard work on the refactor.

**Pull request merged**